tests/fixtures/up_to_date_lock/pyproject.toml

```diff
@@ -6,7 +6,7 @@ authors = ["Poetry Developer <developer@python-poetry.org>"]
 
 [tool.poetry.dependencies]
 python = "^3.8"
-docker = "4.3.1"
+docker = ">=4.3.1"
```

Comment:
This was necessary for `test_add_does_not_update_locked_dependencies` to let the solver try to (wrongly) update from `4.3.1` to `4.3.2` when `--lock` is specified. I verified that the test was failing with the previous implementation and passes now